File: day-09/part-2/badouralix.py
from runners.python import Submission

class BadouralixSubmission(Submission):

	def run(self, s):
		# :param s: input in string format
		# :return: solution flag
		counter = 0
		garbage = False
		skip = False

		for char in s:
			if skip:
				skip = False
			else:
				if char == '!':
					skip = True
				elif char == '>':
					garbage = False
				elif not garbage and char == '<':
					garbage = True
				elif garbage:
					counter += 1

		return counter

File: day-09/part-2/badouralix.c
#include <stdio.h>
#include <stdbool.h>

const int run(char* s)
{
    int counter = 0;
    bool garbage = false;

    for (char* c = --s; *++c != '\0';) {
        if (garbage) {
            switch (*c) {
            case '!':
                c++;
                break;
            case '>':
                garbage = false;
                break;
            default:
                counter++;
                break;
            }
        } else {
            switch (*c) {
            case '<':
                garbage = true;
                break;
            default:
                break;
            }
        }
    }

    return counter;
}

int main(int argc, char** argv)
{
    printf("%d\n", run(argv[1]));
    return 0;
}
